packages/shared/src/domain/address/state.ts

```diff
@@ -1,17 +1,15 @@
+import { BadRequestError } from "../../error/bad-request";
+
 export function normalizeStateSafe(state: string): USState | undefined {
-  if (Object.values(states).includes(USState[state as keyof typeof USState])) {
-    return USState[state as keyof typeof USState];
-  } else if (states[state]) {
-    return states[state];
-  } else if (state === "DC") {
-    return USState.DC;
-  }
-  return undefined;
+  const keyFromEntries = Object.entries(states).find(
+    ([key, value]) => key === state || value.toLowerCase() === state.toLowerCase()
+  );
+  return keyFromEntries?.[0] as USState | undefined;
 }
 
 export function normalizeState(state: string): USState {
   const stateOrUndefined = normalizeStateSafe(state);
-  if (!stateOrUndefined) throw new Error("Invalid state.");
+  if (!stateOrUndefined) throw new BadRequestError("Invalid state", undefined, { state });
```

Comment:
As noted: normalization *doesn't always* map to user submitted input, but default to BadRequetError is fine as we can consider the default case to be user submitted.